packages/react-devtools-shared/src/backend/DevToolsComponentStackFrame.js

```diff
@@ -0,0 +1,298 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+// This is a DevTools fork of ReactComponentStackFrame.
+// This fork enables DevTools to use the same "native" component stack format,
+// while still maintaining support for multiple renderer versions
+// (which use different values for ReactTypeOfWork).
+
+import type {Source} from 'shared/ReactElementType';
+import type {LazyComponent} from 'react/src/ReactLazy';
+import type {CurrentDispatcherRef} from './types';
+
+import {
+  BLOCK_NUMBER,
+  BLOCK_SYMBOL_STRING,
+  FORWARD_REF_NUMBER,
+  FORWARD_REF_SYMBOL_STRING,
+  LAZY_NUMBER,
+  LAZY_SYMBOL_STRING,
+  MEMO_NUMBER,
+  MEMO_SYMBOL_STRING,
+  SUSPENSE_NUMBER,
+  SUSPENSE_SYMBOL_STRING,
+  SUSPENSE_LIST_NUMBER,
+  SUSPENSE_LIST_SYMBOL_STRING,
+} from './ReactSymbols';
+
+// These methods are safe to import from shared;
+// there is no React-specific logic here.
+import {disableLogs, reenableLogs} from 'shared/ConsolePatchingDev';
+
+let prefix;
+export function describeBuiltInComponentFrame(
+  name: string,
+  source: void | null | Source,
+  ownerFn: void | null | Function,
+): string {
+  if (prefix === undefined) {
+    // Extract the VM specific prefix used by each line.
+    try {
+      throw Error();
```

Comment:
nit: i don't think there is much need to throw this error, `.stack` is synchronously readable

Reply:
This is a left over from the upstream implementation where it is required because Internet Explorer requires that you throw the error to assign a stack to it.

In the fork it's not technically needed since we don't support Internet Explorer with DevTools but we could in theory have some user space hosted DevTools that could work in such environments.

Reply:
FWIW I think the iframe version of DevTools does actually support IE (last I tried it)